htmlelements-matchers/src/main/java/ru/yandex/qatools/htmlelements/matchers/decorators/ConditionMatcherDecorator.java

```diff
@@ -22,16 +25,22 @@ public ConditionMatcherDecorator(Matcher<? super T> matcher, E itemToMatchCondit
     protected boolean matchesSafely(T item) {
         if (condition.matches(itemToMatchCondition)) {
             return matcher.matches(item);
+        } else {
```

Comment:
agree with StringDescription chaining, but we prefer "guard if" than if-else. So you can just skip some code if condition is false.